src/vscode-dts/vscode.proposed.chatParticipantAdditions.d.ts

```diff
@@ -240,6 +240,17 @@ declare module 'vscode' {
 		 * Provide a set of variables that can only be used with this participant.
 		 */
 		participantVariableProvider?: { provider: ChatParticipantCompletionItemProvider; triggerCharacters: string[] };
+
+		/**
+		 * Event that fires when a request is paused or unpaused.
+		 * Chat requests are initialy unpaused in the {@link requestHandler}.
+		 */
+		onDidChangePauseState: Event<ChatParticipantPauseStateEvent>;
```

Comment:
@connor4312 curious why you choose this global approach over something that's passed to the request handler?

Reply:
Just since none of the existing data passed there was 'live', but I'm more than happy to change this :)

Reply:
The cancellation token could be something. Today, it's very special and handled by the RPC implementation but we could have something similar or a combination of pause and cancel

Reply:
Are you thinking something like

```ts
class PausableToken extends CancellationToken {
  onDidChangePauseState: Event<boolean>;
  isPaused: boolean;
}
```

?

Reply:
Yeah, something along those lines